### PR TITLE
Upgrade to .NET Core 3, fixes #40

### LINKED ISSUE #40: Upgrade to .NET Core 3

### PR DESCRIPTION
**Pull request by Suchiman**

### CODE REVIEW

**Comment by Suchiman** [comment]
I guess i'll postpone this until ~3rd December where .NET Core 3.1 will (likely) release

**Comment by kog** [comment]
@Suchiman - sure, sounds legit. As an aside, would you mind seeing if you can push a feature branch to the main channel repo? I wanna make sure that you have all the permissions you might need to do that.

File: src/SpikeCore/SpikeCore/Modules/FactoidModule.cs [hunk 1]
```diff
@@ -56,31 +55,35 @@ protected override async Task HandleMessageAsyncInternal(IrcPrivMessage request,
...
                         var paginationDisplay = (count > FactDisplayCount)
-                            ? $" (showing the first {Math.Min(count, FactDisplayCount)})"
+                            ? $" (showing the first {FactDisplayCount})"
                             : string.Empty;
-                        
+
                         var factoidDisplay = matchingFactoids
-                            .Take(FactDisplayCount)
-                            .Select(x => $"[{x.Description} at {x.CreationDate:MM/dd/yyyy H:mm:ss UTC} by {x.CreatedBy}]")
-                            .Join();
```
Comment (Suchiman) [inline src/SpikeCore/SpikeCore/Modules/FactoidModule.cs]: I took the freedom to replace the EF internal API that is causing "unstable API" warnings by just String.Join 😄

File: src/SpikeCore/SpikeCore/Modules/FactoidModule.cs [hunk 2]
```diff
@@ -56,31 +55,35 @@ protected override async Task HandleMessageAsyncInternal(IrcPrivMessage request,
...
+                        // They'll have to hit the web UI for these.
+                        if (count == FactDisplayCount)
+                        {
+                            count = query.Count();
+                        }
+
                         var pluralization = (count == 1) ? string.Empty : "s";
 
                         var paginationDisplay = (count > FactDisplayCount)
-                            ? $" (showing the first {Math.Min(count, FactDisplayCount)})"
```
Comment (Suchiman) [inline src/SpikeCore/SpikeCore/Modules/FactoidModule.cs]: My Math skills suck but i think it's a given that when `count > FactDisplayCount` then we can just display `FactDisplayCount`

Comment (kog) [reply src/SpikeCore/SpikeCore/Modules/FactoidModule.cs]: This is most likely copy/paste from the last implementation, but generally speaking I just throw stuff like that in code out of habit. I guess it's probably unnecessary.

File: src/SpikeCore/SpikeCore/Modules/FactoidModule.cs (hunk 1 above)
Comment (kog) [reply src/SpikeCore/SpikeCore/Modules/FactoidModule.cs]: Ah, interesting. Thanks for doing this.

File: src/SpikeCore/SpikeCore/Modules/FactoidModule.cs [hunk 3]
```diff
@@ -56,31 +55,35 @@ protected override async Task HandleMessageAsyncInternal(IrcPrivMessage request,
...
                 else
                 {
-                    // Grab the total count so we can tell users if there's more facts they can't see. They'll have to
-                    // hit the web UI for these.
-                    var count = _context.Factoids
-                        .Count(factoid => FactoidMatches(factoid, command, name));                   
+                    var query = _context.Factoids
+                        .Where(factoid => factoid.Type.ToLower() == command.ToLower() && factoid.Name.ToLower() == name.ToLower());
 
+                    var matchingFactoids = query
```
Comment (kog) [inline src/SpikeCore/SpikeCore/Modules/FactoidModule.cs]: Yeah, I guess this is probably cleaner than the copy pasta we ported over from the last bot. Thanks for the refactor.

**Pull request merged**